File: Data Structures & Algorithms/max-water-container/submission-0.cs
public class Solution {
        public int MaxArea(int[] height)
    {

        int max = 0;
        for (int i = 0; i < height.Length; i++)
        {
            Console.WriteLine("------------------------->");
            for (int j = i+1; j < height.Length; j++)
            {
                var minlocal = int.Min(height[i], height[j])*(j-i);
                if(minlocal > max)
                    max = minlocal;

            }
                
        }
        return max;
    }
}
    

File: Data Structures & Algorithms/max-area-of-island/submission-0.java
class Solution {
    int ROWS=0;
    int COLS=0;
    public int maxAreaOfIsland(int[][] grid) {
        ROWS= grid.length;
        COLS=grid[0].length;
        int count=0;
        for(int i=0;i<ROWS;i++){
            for (int j=0;j<COLS;j++){
                if(grid[i][j]==1)
                    count=Math.max(count,dfs(grid,i,j));
            }
        }
        return count;
    }

    public int dfs(int[][] grid,int x,int y){
        if(Math.min(x,y)<0 ||
                x>=ROWS ||
                y>=COLS ||
                grid[x][y]==0
        )
            return 0;
        grid[x][y]=0;
        return 1+dfs(grid,x+1,y)+dfs(grid,x-1,y)+dfs(grid,x,y+1)+dfs(grid,x,y-1);
    }
}
